manifests/manifests.go

```diff
@@ -29,19 +28,9 @@ type Manifests interface {
 	CreateManifestPatch(originalManifests, modifiedManifests []byte, originalSource, modifiedSource string) ([]byte, error)
 	// ApplyManifestPatch applies a manifest patch (obtained with CreateManifestDiff) returned the patched manifests
 	ApplyManifestPatch(originalManifests, patchManifests []byte, originalSource, patchSource string) ([]byte, error)
-}
-
-func AppendManifestToBuffer(manifest []byte, buffer *bytes.Buffer) error {
-	separator := "---\n"
-	bytes := buffer.Bytes()
-	if len(bytes) > 0 && bytes[len(bytes)-1] != '\n' {
-		separator = "\n---\n"
-	}
-	if _, err := buffer.WriteString(separator); err != nil {
-		return fmt.Errorf("cannot write to internal buffer: %s", err)
-	}
-	if _, err := buffer.Write(manifest); err != nil {
-		return fmt.Errorf("cannot write to internal buffer: %s", err)
-	}
-	return nil
+	// AppendManifestToBuffer concatentates manifest bytes to a
+	// (possibly empty) buffer of manifest bytes; the resulting bytes
+	// should be parsable by `ParseManifest`.
+	// TODO(michael) should really be an interface rather than `*bytes.Buffer`.
```

Comment:
Agreed, can we make it a writer and rename it to `AppendManifest`?

(To be fair all the `[]byte`'s  should probably be readers and writers too)

Reply:
I did give it a quick attempt before writing the comment, but there were complications -- I can't remember what they were :-(. Possibly just that the implementation uses methods of `*bytes.Buffer`.